src/expire.c

```diff
@@ -268,10 +286,16 @@ void activeExpireCycle(int type) {
              * than keys in the same time. */
             long max_buckets = num*20;
             long checked_buckets = 0;
-
+            
             while (data.sampled < num && checked_buckets < max_buckets) {
-                db->expires_cursor = dictScan(db->expires, db->expires_cursor,
+                int slot = getAndClearSlotIdFromCursor((unsigned long long *)&db->expires_cursor);
```

Comment:
maybe we can create a dbScan() interface, instead of messing with the cursor here?
imagine we'll need such scan in another place (or a few others) in the future, i don't like to keep cloning that logic.
IIRC we do that in module.c too already.